### PR TITLE
Feature#48

### PR DESCRIPTION
**Pull request by hamzakhawaja27**
@jamesfoote Please check if the structure of the method is similar to the one you are doing for GKE.

### CODE REVIEW

File: src/ucapi/ecs_client.py [hunk 1]
```diff
@@ -498,3 +498,32 @@ def get_api_client(self):
...
+               :param task_id: The id of the task to be replaced. str
+
+               :param task definition of the task which replaces the old task. dict
+
+               """
+
+
+
+        self._aws_ecs_client.stop_task(cluster=cluster_name,\
+                             task= task_id)
```
Comment (BlairTwiss) [inline src/ucapi/ecs_client.py]: Why are these on a separate line?

File: src/ucapi/ecs_client.py [hunk 2]
```diff
@@ -498,3 +498,32 @@ def get_api_client(self):
...
+               with the same name but a different version number.
+
+               :param cluster_name: The name of the cluster in which the task is to be updated. str
+
+               :param task_id: The id of the task to be replaced. str
+
+               :param task definition of the task which replaces the old task. dict
+
+               """
+
```
Comment (BlairTwiss) [inline src/ucapi/ecs_client.py]: Too many blank lines, should only be one

File: src/ucapi/ecs_client.py [hunk 3]
```diff
@@ -498,3 +498,32 @@ def get_api_client(self):
...
+
+    def update_task(self, task_id, cluster_name, task_definition):
+        """
+               replaces the task running on a current cluster. The only way of updating a task is to create a revision which is essentially a new task
+               with the same name but a different version number.
+
+               :param cluster_name: The name of the cluster in which the task is to be updated. str
+
+               :param task_id: The id of the task to be replaced. str
```
Comment (BlairTwiss) [inline src/ucapi/ecs_client.py]: Don't need blank lines between parameters

File: src/ucapi/ecs_client.py [hunk 4]
```diff
@@ -498,3 +498,32 @@ def get_api_client(self):
...
+
+               :param task definition of the task which replaces the old task. dict
+
```
Comment (BlairTwiss) [inline src/ucapi/ecs_client.py]: Add what function returns

Comment (SukritGupta17) [reply src/ucapi/ecs_client.py]: also make the doc string google style.. like this:
Args:
        task_id:
Returns:

File: src/ucapi/ecs_client.py [hunk 5]
```diff
@@ -498,3 +498,26 @@ def get_api_client(self):
...
+    def update_task(self, task_id, cluster_name, task_definition):
+        """
+               replaces the task running on a current cluster. The only way of updating a task is to create a revision which is essentially a new task
+               with the same name but a different version number.
+               Args:
+               cluster_name: The name of the cluster in which the task is to be updated. str
+               task_id: The id of the task to be replaced. str
+               task definition: of the task which replaces the old task. dict
+               Returns:
+               response: response from the AWS server.
```
Comment (SukritGupta17) [inline src/ucapi/ecs_client.py]: add indentation to line 508 509 510 512
add a empty line after 506 and 510

File: README.md [hunk 6]
```diff
@@ -37,6 +37,8 @@ Blair Twiss - u5673405@anu.edu.au
 
 James Foote - u5370251@anu.edu.au
 
+Hamza Khawaja - u6019739@anu.edu.au
+
```
Comment (SukritGupta17) [inline README.md]: no need, already added the name

**Pull request closed without merging**